app/src/main/java/com/example/project/MapsFragment.kt

```diff
@@ -132,49 +269,202 @@ class MapsFragment : Fragment(), OnMapReadyCallback {
+                            )
+
+                            // get nearby places/restaurants
+                            val position = "${lastKnownLocation!!.latitude},${lastKnownLocation!!.longitude}"
+                            viewModel.getNearbySearch(position,Constants.RADIUS_1000,Constants.TYPE,getString(R.string.google_maps_key))
+                            viewModel.nearbySearchresponse.observe(this, { response ->
+                                Log.d(Constants.TAG,"${response.results}")
+                                markNearbyPlacesOnMap(response.results)
+                            })
+
+                            map?.moveCamera(
+                                CameraUpdateFactory.newLatLngZoom(
+                                    cLocation, Constants.DEFAULT_ZOOM.toFloat()
+                                )
+                            )
+
+                            BitmapDescriptorFactory.fromResource(R.drawable.house)
+                            map?.addMarker(
+                                MarkerOptions().position(cLocation).title(
+                                    getAddress(
+                                        cLocation.latitude,
+                                        cLocation.longitude
+                                    )
+                                ).icon(BitmapDescriptorFactory.fromBitmap(bitResized))
+                            )
                         }
                     } else {
-                        Log.d(TAG, "Current location is null. Using defaults location.")
-                        Log.e(TAG, "Exception: %s", task.exception)
-                        map?.moveCamera(CameraUpdateFactory
-                            .newLatLngZoom(defaultLocation, DEFAULT_ZOOM.toFloat()))
-                        map?.addMarker(MarkerOptions().position(defaultLocation).title(getAddress(defaultLocation.latitude,defaultLocation.longitude)))
+                        Log.d(Constants.TAG, "Current location is null. Using defaults location.")
+                        Log.e(Constants.TAG, "Exception: %s", task.exception)
+                        map?.moveCamera(
+                            CameraUpdateFactory
+                                .newLatLngZoom(defaultLocation, Constants.DEFAULT_ZOOM.toFloat())
+                        )
+                        map?.addMarker(
+                            MarkerOptions().position(defaultLocation).title(
+                                getAddress(
+                                    defaultLocation.latitude,
+                                    defaultLocation.longitude
+                                )
+                            ).icon(BitmapDescriptorFactory.fromBitmap(bitResized))
+                        )
                         map?.uiSettings?.isMyLocationButtonEnabled = false
                     }
+                    //set marker anywhere clicked on the map
+                    map?.setOnMapClickListener{
```

Comment:
I suggest commenting out this. Right now there's no purpose for immediately creating a marker at the clicked point. This, however, may be useful later